src/v/kafka/client/configuration.h

```diff
@@ -37,6 +37,7 @@ struct configuration final : public config::config_store {
     config::property<std::chrono::milliseconds> produce_shutdown_delay;
     config::property<int16_t> produce_ack_level;
     config::property<std::chrono::milliseconds> consumer_request_timeout;
+    config::property<int32_t> consumer_request_min_bytes;
```

Comment:
for my own edification: why is this signed?

Reply:
It's passed fairly directly to a POD representing the request, and the serialised types are int32.

Reply:
> serialised types are int32

ah, right. 👍

Reply:
> we would see a different error in the client?

From the logs, this was mostly the error that was tripping up Schema Registry.

Reply:
Ya, what I'm curious about is how that min bytes value is acted on by the fetcher, but I can just refer to the source.

Reply:
It's worth checking if the behaviour is correct.

If I set:
```yaml
schema_registry_client:
  consumer_request_min_bytes: 0
  consumer_request_max_bytes: 0
```
I get back an empty batch, reproducing the issue to some degree, but if I set max greater than 0, I don't see it. My system is not under any load.

Reply:
This looks a little odd to me; it may be semantics or naming, but they don't seem to align.
https://github.com/redpanda-data/redpanda/blob/v23.3.1-rc3/src/v/kafka/server/handlers/fetch.h#L110-L119

Reply:
Yeah that looks a little strange in the context of [fetch_topic_partitions](https://github.com/redpanda-data/redpanda/blob/70f503c4510e735cf4008121bb73b9bccfdd9cf9/src/v/kafka/server/handlers/fetch.cc#L849-L871) . Certainly if `min_bytes == 0` it seems sort of nonsensical, as in "we tried, got nothing, and that was enough, so 👍".

Reply:
If I'm reading this correctly, I think an empty fetch will trigger a retry of some sort, assuming min_bytes > 0.